src/main.ts

```diff
@@ -17,8 +17,17 @@ export class ProperLogger {
 
   console: Pick<Console, 'debug' | 'log' | 'warn' | 'error'>;
 
+  static loggerInstances: Record<string, ProperLogger> = {};
```

Comment:
Given our intended usage (mostly Lambda-based APIs that have short lifetime, mostly using the same "name" consistently vs. many different names) I think we don't really need to worry about this cache growing too large and never being cleared (which would otherwise be a bit worrisome in other cases).

Reply:
Yep. Good point. That would also require many different names being used. And maybe "dynamically generated names" for truly "unbounded growth". If we were in a situation where we needed that we could potentially use the `constructor` rather than this static method.